src/gluonts/evaluation/_base.py

```diff
@@ -196,28 +237,21 @@ def seasonal_error(
         where m is the seasonal frequency
         https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
         """
-        # Remove the prediction range
-        # If the prediction range is not in the end of the time series,
-        # everything after the prediction range is truncated
-        forecast_date = pd.Timestamp(forecast.start_date, freq=forecast.freq)
-        date_before_forecast = forecast_date - 1 * forecast_date.freq
-        ts = time_series[:date_before_forecast]
-
         # Check if the length of the time series is larger than the seasonal frequency
         seasonality = (
             self.seasonality
             if self.seasonality
             else get_seasonality(forecast.freq)
         )
-        if seasonality < len(ts):
+        if seasonality < len(past_data):
             forecast_freq = seasonality
         else:
             # edge case: the seasonal freq is larger than the length of ts
             # revert to freq=1
             # logging.info('The seasonal frequency is larger than the length of the time series. Reverting to freq=1.')
             forecast_freq = 1
-        y_t = np.ma.masked_invalid(ts.values[:-forecast_freq])
-        y_tm = np.ma.masked_invalid(ts.values[forecast_freq:])
+        y_t = past_data[:-forecast_freq]
+        y_tm = past_data[forecast_freq:]
```

Comment:
Why removing the masking here?

Reply:
Becaus `past_data` is already masked, so it is now not necessary to mask again. Previously this was not the case.